### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version stable/2019-03-22

### PR DESCRIPTION
**Pull request by reshantchandra**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/reshantchandra/azure-rest-api-specs/dev-iothub-Microsoft.Devices-2019-03-22...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json [hunk 1]
```diff
@@ -0,0 +1,3482 @@
...
+    "IotHubProperties": {
+      "description": "The properties of an IoT hub.",
+      "type": "object",
+      "properties": {
+        "authorizationPolicies": {
+          "description": "The shared access policies you can use to secure a connection to the IoT hub.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+
```
Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]: We need to make sure this isn't bugged

Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]:
```suggestion
          "description": "Specific role assigned to this location",
```

Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]:
```suggestion
         "description": "Azure region",
```

**Review by jlian — CHANGES_REQUESTED** [review]

**Comment by jlian** [comment]
Reason why we closed the PR for now: just talked to @reshantchandra and we noticed that the `locations` property isn't being properly returned by IoT Hub service for this API version (unlike `2019-03-22-preview`). We figured the best thing to do is to close the PR until we figure out what to do.

The part that's missing

```json
    "locations": [
      {
        "location": "West US",
        "role": "primary"
      },
      {
        "location": "East US",
        "role": "secondary"
      }
    ],
```

**Comment by reshantchandra** [comment]
We resolved the location property issue now we see primary and secondary location as expected.

**Review by shansin — APPROVED** [review]

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json [hunk 2]
```diff
@@ -0,0 +1,3482 @@
...
+          "description": "The certificate content",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "IotHubProperties": {
+      "description": "The properties of an IoT hub.",
+      "type": "object",
+      "pr
```
Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]:
```suggestion
          "description": "The role of the region, can be either primary or secondary. The primary region is where the IoT hub is currently provisioned. The secondary region is the the Azure disaster recovery (DR) paired region and also the region where the IoT hub can failover to.",
```

Comment (jlian) [inline specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]:
```suggestion
         "description": "The name of the Azure region",
```

**Review by jlian — CHANGES_REQUESTED** [review]
Small description update

File: specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json (hunk 2 above)
Comment (jlian) [reply specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]: Wait, looks like this feature is buggy

Comment (jlian) [reply specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]: @reshantchandra don't use the commit suggestion feature, I think it's going to modify the wrong line

Comment (jlian) [reply specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]: Please update the json directly

Comment (nschonni) [reply specification/iothub/resource-manager/Microsoft.Devices/stable/2019-03-22/iothub.json]: The patch is probably right. GitHub just messes up these line comments when it comes to large files. It usually looks OK on the files tab

**Review by anthony-c-martin — APPROVED** [review]

**Comment by yungezz** [comment]
hi @reshantchandra  @MyronFanQiu what's the status of this PR? is it ready for merge?

**Review by mmyyrroonn — APPROVED** [review]
LGTM. Compared with the corresponding preview version. Just small change. Please solve the ARM's comment and I can merge this PR.

**Comment by MyronFanQiu** [comment]
@reshantchandra Please resolve the ARM's comments. @jlian Looks like the author has updated the PR. Could you please check it and approve this PR?

**Comment by reshantchandra** [comment]
comments resolved

**Comment by MyronFanQiu** [comment]
@jlian If you think this PR is good, please approve it so that I can merge this PR.

**Review by jlian — APPROVED** [review]

**Pull request merged**